parts/k8s/kubernetesagentcustomdata.yml

```diff
@@ -92,15 +92,17 @@ write_files:
     TimeoutStartSec=0
     Restart=on-failure
     RestartSec=5s
-    ExecStartPre=/bin/mkdir -p /tmp/kubectldir
+    ExecStartPre=/bin/mkdir -p /tmp/hyperkubedir
     ExecStartPre=/usr/bin/docker pull {{WrapAsVariable "kubernetesHyperkubeSpec"}}
-    ExecStartPre=/usr/bin/docker run --rm -v /tmp/kubectldir:/opt/kubectldir {{WrapAsVariable "kubernetesHyperkubeSpec"}} /bin/bash -c "cp /hyperkube /opt/kubectldir/"
+    ExecStartPre=/usr/bin/docker run --rm -v /tmp/hyperkubedir:/opt/hyperkubedir {{WrapAsVariable "kubernetesHyperkubeSpec"}} /bin/bash -c "cp /hyperkube /opt/hyperkubedir/"
```

Comment:
Besides the extraction, dependencies should also be installed. e.g. on ubuntu, kubelet depends on `iptables (>= 1.4.21), kubernetes-cni (= 0.6.0), iproute2, socat, util-linux, mount, ebtables, ethtool, init-system-helpers (>= 1.18~)`

Reply:
Thanks @feiskyer. Just checked and all of these are already being installed except for `socat`. I will add a step to install them if they are missing so we are safe. I can't find a package `kubernetes-cni` however, do you have more info on that one?

Reply:
For the cni plugin, I think we are already installing it at https://github.com/Azure/acs-engine/blob/b463cd0b0e5818e9da87ef75a86dc45ac14dc014/parts/k8s/kubernetesmastercustomscript.sh#L217